main.go

```diff
@@ -78,6 +81,7 @@ var (
 
 func main() {
 	initFlags(pflag.CommandLine)
+	pflag.CommandLine.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
```

Comment:
Why do we need this line? I don't see it in the original suggestion, but i accept that maybe we just never added it and need it now?

Reply:
It just allows you to add a function which can translate flag names

Reply:
https://github.com/kubernetes/component-base/blob/master/cli/flag/flags.go#L37

Do we have flags (or inherit from upstream) flags that contain underscore? This would change them to dashes. 

Where is this recommendation offered in upstream docs, templates, or ecosystem providers?

Reply:
I see that the upstream uses this option here: https://github.com/kubernetes-sigs/cluster-api/blob/main/main.go#L229

I'm not sure if we need it though.